### ISSUE TITLE
[JENKINS-52825] Cannot see executors in Blue Ocean UI

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#4811]
<p>In the Classic UI, it is possible to see the <b>Build Executor Status</b></p>

<p>This view is not available in Blue Ocean.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kzantow" class="user-hover" rel="kzantow">kzantow</a> worked on this plugin:  <a href="https://github.com/johnnyq1313/blueocean-executor-info-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/johnnyq1313/blueocean-executor-info-plugin</a></p>

<p>Please finish off this plugin and make it part of the default <b>Update Center</b></p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rodrigc.jpg" title="rodrigc's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rodrigc">rodrigc</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-52825" target="_blank">Cannot see executors in Blue Ocean UI</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale">michaelneale</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Minor
<li><b>component(s)</b>: blueocean-executor-info-plugin, blueocean-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2019-03-02T07:59:12+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>In the Classic UI, it is possible to see the <b>Build Executor Status</b></p>

<p>This view is not available in Blue Ocean.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kzantow" class="user-hover" rel="kzantow">kzantow</a> worked on this plugin:  <a href="https://github.com/johnnyq1313/blueocean-executor-info-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/johnnyq1313/blueocean-executor-info-plugin</a></p>

<p>Please finish off this plugin and make it part of the default <b>Update Center</b></p></pre>
</details>
</details>
<details><summary><i>2 attachments</i></summary>

- [Screen Shot 2018-08-21 at 8.21.47 AM.png](https://issues.jenkins.io/secure/attachment/43809/Screen%20Shot%202018-08-21%20at%208.21.47%20AM.png)
  > ![Screen Shot 2018-08-21 at 8.21.47 AM.png](https://issues.jenkins.io/secure/attachment/43809/Screen%20Shot%202018-08-21%20at%208.21.47%20AM.png)
- [screenshot-1.png](https://issues.jenkins.io/secure/attachment/43828/screenshot-1.png)
  > ![screenshot-1.png](https://issues.jenkins.io/secure/attachment/43828/screenshot-1.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rodrigc.jpg" title="rodrigc's avatar" /> rodrigc:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-52825?focusedId=345174&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-345174">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>On Gitter, <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale" class="user-hover" rel="michaelneale">michaelneale</a> mentioned that instead of blueocean-executor-info-plugin being a separate plugin,</p>

<p>the functionality of displaying executors could be incorporated into the blueocean core plugin.</p>

<p> </p>

<p>It doesn't matter to me if this is in core or a separate plugin.  Whatever you decide is fine.</p>
</pre>
</details>

</ul>
</details>

<p>On Gitter, <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale" class="user-hover" rel="michaelneale">michaelneale</a> mentioned that instead of blueocean-executor-info-plugin being a separate plugin,</p>

<p>the functionality of displaying executors could be incorporated into the blueocean core plugin.</p>

<p> </p>

<p>It doesn't matter to me if this is in core or a separate plugin.  Whatever you decide is fine.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-52825?focusedId=345249&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-345249">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Cc <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jbriden" class="user-hover" rel="jbriden">jbriden</a></p>
</pre>
</details>

</ul>
</details>

<p>Cc <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jbriden" class="user-hover" rel="jbriden">jbriden</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-52825?focusedId=345250&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-345250">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>BTW there is probably enough functionality in <a href="https://github.com/jenkinsci/blueocean-executor-info-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-executor-info-plugin</a> - if someone was to do a PR which rolled that into dashboard - maybe no one would object... </p>
</pre>
</details>

</ul>
</details>

<p>BTW there is probably enough functionality in <a href="https://github.com/jenkinsci/blueocean-executor-info-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-executor-info-plugin</a> - if someone was to do a PR which rolled that into dashboard - maybe no one would object... </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rodrigc.jpg" title="rodrigc's avatar" /> rodrigc:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-52825?focusedId=345254&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-345254">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I presume you mean merge the functionality of <a href="https://github.com/jenkinci/blueocean-executor-info-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinci/blueocean-executor-info-plugin</a> into <a href="https://github.com/jenkinsci/blueocean-plugin/tree/master/blueocean-dashboard" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/tree/master/blueocean-dashboard</a> ?</p>
</pre>
</details>

</ul>
</details>

<p>I presume you mean merge the functionality of <a href="https://github.com/jenkinci/blueocean-executor-info-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinci/blueocean-executor-info-plugin</a> into <a href="https://github.com/jenkinsci/blueocean-plugin/tree/master/blueocean-dashboard" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/tree/master/blueocean-dashboard</a> ?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rodrigc.jpg" title="rodrigc's avatar" /> rodrigc:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-52825?focusedId=345256&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-345256">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I checked out the code for <a href="https://github.com/jenkinci/blueocean-executor-info-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinci/blueocean-executor-info-plugin</a>  , built it with <b>mvn package</b>, and then installed <b>blueocean-executor-info-plugin.hpi</b> .  However, when I switched to the Blue Ocean UI, it was blank.  In the Chrome Inspector, there were errors:</p>

<p> </p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>Module load failure: Timed out waiting on module 'blueocean-executor-info:jenkins-js-extension' to load.
(anonymous) @ blueocean-core-js.js:2295
jenkins-js-extension.js:691 Module load failure: Timed out waiting on module 'jenkins-cd-blueocean-core-js-dist-js-i18n-bundle-startup:jenkins-cd-blueocean-core-js-dist-js-i18n-bundle-startup' to load.
(anonymous) @ jenkins-js-extension.js:691</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>I checked out the code for <a href="https://github.com/jenkinci/blueocean-executor-info-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinci/blueocean-executor-info-plugin</a>  , built it with <b>mvn package</b>, and then installed <b>blueocean-executor-info-plugin.hpi</b> .  However, when I switched to the Blue Ocean UI, it was blank.  In the Chrome Inspector, there were errors:</p>

<p> </p>


```
Module load failure: Timed out waiting on module 'blueocean-executor-info:jenkins-js-extension' to load.
(anonymous) @ blueocean-core-js.js:2295
jenkins-js-extension.js:691 Module load failure: Timed out waiting on module 'jenkins-cd-blueocean-core-js-dist-js-i18n-bundle-startup:jenkins-cd-blueocean-core-js-dist-js-i18n-bundle-startup' to load.
(anonymous) @ jenkins-js-extension.js:691
```

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rodrigc.jpg" title="rodrigc's avatar" /> rodrigc:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-52825?focusedId=346664&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-346664">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kzantow" class="user-hover" rel="kzantow">kzantow</a> do you have a screenshot of what <a href="https://github.com/jenkinci/blueocean-executor-info-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinci/blueocean-executor-info-plugin</a>  looked like?</p>

<p>If you don't have enough cycles, I'd like to dust this code off, and submit a PR to merge it into the blueocean-dashboard in the blueocean-plugin</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kzantow" class="user-hover" rel="kzantow">kzantow</a> do you have a screenshot of what <a href="https://github.com/jenkinci/blueocean-executor-info-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinci/blueocean-executor-info-plugin</a>  looked like?</p>

<p>If you don't have enough cycles, I'd like to dust this code off, and submit a PR to merge it into the blueocean-dashboard in the blueocean-plugin</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kzantow.png" title="kzantow's avatar" /> kzantow:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-52825?focusedId=346717&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-346717">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rodrigc" class="user-hover" rel="rodrigc">rodrigc</a> unfortunately, the visual design is not good enough to merge it into BO proper - it was just a proof of concept that extensions work (it is/was also used to verify that nothing broke extensions in the Blue Ocean ATH). I'll update it to the latest BO release version. You can run the plugin by just:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">git checkout ...
mvn install
mvn hpi:run
</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rodrigc" class="user-hover" rel="rodrigc">rodrigc</a> unfortunately, the visual design is not good enough to merge it into BO proper - it was just a proof of concept that extensions work (it is/was also used to verify that nothing broke extensions in the Blue Ocean ATH). I'll update it to the latest BO release version. You can run the plugin by just:</p>

<pre>
git checkout ...
mvn install
mvn hpi:run
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rodrigc.jpg" title="rodrigc's avatar" /> rodrigc:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-52825?focusedId=346732&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-346732">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/43809/43809_Screen+Shot+2018-08-21+at+8.21.47+AM.png" style="border: 0px solid black" /></span></p>
</pre>
</details>

</ul>
</details>

<p><span class="image-wrap" style=""><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/43809/Screen%20Shot%202018-08-21%20at%208.21.47%20AM.png" style="border: 0px solid black" /></span></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rodrigc.jpg" title="rodrigc's avatar" /> rodrigc:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-52825?focusedId=346733&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-346733">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I was able to run this plugin via <b>mvn hpi:run</b></p>

<p>The UI could be improved a bit, but it does work in terms of providing a view of executors inside Blue Ocean.</p>

<p> </p>

<p>It would be nice of this plugin could be enhanced and merged into the blueocean-dashboard.</p>

<p> </p>

<p>That would eliminate the need to switch back to the classic UI to view the status of executors.</p>

<p> </p>
</pre>
</details>

</ul>
</details>

<p>I was able to run this plugin via <b>mvn hpi:run</b></p>

<p>The UI could be improved a bit, but it does work in terms of providing a view of executors inside Blue Ocean.</p>

<p> </p>

<p>It would be nice of this plugin could be enhanced and merged into the blueocean-dashboard.</p>

<p> </p>

<p>That would eliminate the need to switch back to the classic UI to view the status of executors.</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rodrigc.jpg" title="rodrigc's avatar" /> rodrigc:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-52825?focusedId=346743&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-346743">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Even though the UI is not perfect, even if it was merged as-is into the blueocean-dashboard</p>

<p>and improved iteratively over time, that would be an improvement, and one more</p>

<p>reason why I can use Blue Ocean as my "daily driver" and not have to shift back to the Classic UI.</p>
</pre>
</details>

</ul>
</details>

<p>Even though the UI is not perfect, even if it was merged as-is into the blueocean-dashboard</p>

<p>and improved iteratively over time, that would be an improvement, and one more</p>

<p>reason why I can use Blue Ocean as my "daily driver" and not have to shift back to the Classic UI.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kzantow.png" title="kzantow's avatar" /> kzantow:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-52825?focusedId=346903&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-346903">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Also on the branches tab, executing things show up:  <span class="image-wrap" style=""><a id="43828_thumb" href="https://issues.jenkins.io/secure/attachment/43828/43828_screenshot-1.png" title="screenshot-1.png" file-preview-type="image" file-preview-id="43828" file-preview-title="screenshot-1.png"><img src="https://issues.jenkins.io/secure/thumbnail/43828/_thumb_43828.png" style="border: 0px solid black" role="presentation"/></a></span> </p>
</pre>
</details>

</ul>
</details>

<p>Also on the branches tab, executing things show up:  <span class="image-wrap" style=""><a id="43828_thumb" href="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/43828/screenshot-1.png" title="screenshot-1.png" file-preview-type="image" file-preview-id="43828" file-preview-title="screenshot-1.png"><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/43828/screenshot-1.png" style="border: 0px solid black" role="presentation"/></a></span> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kzantow.png" title="kzantow's avatar" /> kzantow:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-52825?focusedId=346905&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-346905">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rodrigc" class="user-hover" rel="rodrigc">rodrigc</a> This is published to the experimental update center, but needs an update to a recent BO version, as there was a breaking change in Blue Ocean 1.4. Here's a PR to fix it: <a href="https://github.com/jenkinsci/blueocean-executor-info-plugin/pull/4" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-executor-info-plugin/pull/4</a> and we can just publish to the new version to the experimental update center, if that helps you enough.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rodrigc" class="user-hover" rel="rodrigc">rodrigc</a> This is published to the experimental update center, but needs an update to a recent BO version, as there was a breaking change in Blue Ocean 1.4. Here's a PR to fix it: <a href="https://github.com/jenkinsci/blueocean-executor-info-plugin/pull/4" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-executor-info-plugin/pull/4</a> and we can just publish to the new version to the experimental update center, if that helps you enough.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/rodrigc.jpg" title="rodrigc's avatar" /> rodrigc:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-52825?focusedId=361891&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-361891">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Fixed in <a href="https://github.com/jenkinsci/blueocean-plugin/pull/1909" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/1909</a></p>
</pre>
</details>

</ul>
</details>

<p>Fixed in <a href="https://github.com/jenkinsci/blueocean-plugin/pull/1909" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/1909</a></p>
